### ISSUE TITLE
Trouble building on MacOS

### ISSUE THREAD

**Issue by pmetzger** [revery-ui/revery#337]
Disclaimer: I'm an OCaml hacker used to the OPAM ecosystem and I don't know what I'm doing with esy at all.

I've freshly cloned the `revery` repo, I've installed `esy`, I go in to the `revery` directory, and this happens:

```
$ esy install
info install 0.4.10 (using package.json)
error: corrupted esy.lock lock
you might want to remove it and install from scratch
error: error: unknown source type: link-dev
         
  reading lock /Users/perry/src/revery-ui/revery/esy.lock
esy: exiting due to errors above
```

**Comment by rauanmayemir** [comment]
@pmetzger You need a fresher esy, try `npm -g install esy@next`.

**Comment by pmetzger** [comment]
Yes, apparently so. Perhaps advice on this could go into the README?

**Comment by rauanmayemir** [comment]
Actually, latest tag is already on `0.5.6` (as suggested in README). Maybe you had older version?

**Comment by pmetzger** [comment]
It's possible. I'm 100% new to the esy ecosystem.

**Comment by bryphe** [comment]
We debugged this a bit offline - upgrading to esy@0.5.6 helped get a little farther, but we also ran into a build bug finding `-lpng`. We fixed this in revery-ui/reason-fontkit#22 and @pmetzger was kind enough to test out the branch locally - it built successfully after that. 

#338 picks up that fix - so I believe this should be resolved now. Will close out but let me know if there are additional issues. And thanks @rauanmayemir and @anmonteiro for the troubleshooting help!

**Comment by pmetzger** [comment]
It might still be useful to make sure the README says "be absolutely sure you have the latest esy installed" etc.
